doc/proposal/thrift-parameter-validation-proposal.md

```diff
@@ -0,0 +1,187 @@
+| not_nil      | must not be empty                | "true"                               | any                    | -                  |
+| skip         | skip validate                    | "true"                               | any                    | -                  |
+| eq           | equals to (`==`)                 | i8, i16, i32, i64, f64, string, bool | same with value        | -                  |
+| ne           | not equals to (`!=`)             | i8, i16, i32, i64, f64, string, bool | same with value        | -                  |
+| lt           | less than (`<`)                  | i8, i16, i32, i64, f64               | same with value        | -                  |
+| le           | less equal (`<=`)                | i8, i16, i32, i64, f64               | same with value        | -                  |
+| gt           | greater than (`>`)               | i8, i16, i32, i64, f64               | same with value        | -                  |
+| ge           | greater equal (`>=`)             | i8, i16, i32, i64, f64               | same with value        | -                  |
+| in           | within given container           | i8, i16, i32, i64, f64, enum         | same with value        | -                  |
+| not_in       | not within given container       | i8, i16, i32, i64, f64, enum         | same with value        | -                  |
+| elem         | field's element constraint       | any                                  | list, set              | support            |
+| key          | field's element key constraint   | any                                  | map                    | support            |
+| value        | field's element value constraint | any                                  | map                    | support            |
+| min_size     | minimal length                   | i8, i16, i32, i64                    | string, list, set, map | -                  |
+| max_size     | maximal length                   | i8, i16, i32, i64                    | string, list, set, map | -                  |
+| prefix       | field prefix must be             | string                               | string                 | -                  |
+| suffix       | suffix must be                   | string                               | string                 | -                  |
+| contains     | must contain                     | string                               | string                 | -                  |
+| not_contains | must not contain                 | string                               | string                 | -                  |
+| pattern      | basic regular expression         | string                               | string                 | -                  |
+
+- Secodary validator is a successive validator, usually used at container-type field. See below Set/List/Map examples.
```

Comment:
I would prefer to explicitly define `elem` used in examples here as well.

Reply:
> 

I add an explicit definition for commas between rules and also a reference for basic regular expression.